expressions.go

```diff
@@ -538,6 +513,130 @@ func rFirstOf(args *fnArgs) (Value, error) {
 	return vUndefined, nil
 }
 
+type foldNumbers interface {
+	update(value *Number)
+	result() Value
+}
+
+func rFoldNumbers(f foldNumbers, args *fnArgs, min int) (Value, error) {
+	if err := args.checkMinArgsNum(min); err != nil {
+		return nil, err
+	}
+	for i := 0; i < args.num(); i++ {
+		arg, err := args.get(i)
+		if err != nil {
+			return nil, err
+		}
+		switch value := arg.(type) {
+		case *Undefined:
+			return vUndefined, nil
+		case *Number:
+			f.update(value)
+		case *Slice:
+			if value.Len() != 0 {
+				result, err := rFoldNumbers(f, newFnArgs(args.ws, value.Elements()), 0)
+				if err != nil {
+					return nil, err
+				}
+				if result == vUndefined {
+					return vUndefined, nil
+				}
+			}
+		default:
+			return nil, fmt.Errorf("encountered non-numerical argument")
+		}
+	}
+	return f.result(), nil
+}
+
+type sumFolder struct {
+	sum *Number
+}
+
+func (f *sumFolder) update(value *Number) {
+	f.sum = f.sum.Plus(value)
```

Comment:
perhaps allow start with a nil sum like in min/maxFolder? then you wouldn't need to init to vZero

Reply:
if you allow `nil`, then this code would be `if f.sum == nil { f.sum = vZero }`, right? I felt it'd be easier to initialize to `vZero`, plus we want a sum of an empty list to be 0, see https://github.com/helloeave/worksheets/blob/master/computed_by_test.go#L915